src/wscript

```diff
@@ -119,6 +126,67 @@ def build_tools(bld):
             use = ['gaia2'] + bld.env['USELIB'])
 
     # Cyclops server
+    if sys.platform.startswith('linux') and gaia_qt5:
```

Comment:
There's lots of duplicated code. It may be good to shorted the code and assign a features variable first (to ```qt5 qt5network cxx cxxprogram cyclops``` or ```qt4 cxx cxxprogram cyclops``` depending on gaia_qt5) and then run the generic build command.

Reply:
I duplicated the code in the `wscript` files because I can only test on `Ubuntu` and I tried to leave build unchanged for other operating environments - the c++ source code changes likely will work for them too with the `gnu` compiler.  Any changes/improvements recommended are okay with me.  I still have to get the Cyclops option to compile without errors.  The only other potential issue I am aware of is the location of the python directory eg. for 2.7 it is `/usr/local/lib/python2.7/dist-packages`.  For python 3.6 it is `/usr/local/lib/python3/dist-packages` and I had fixed to `/usr/local/lib/python3.6/dist-packages` but wanted that to be more dynamic.

Reply:
Now use conditional `build_tools_qt4` method and `build_tools_qt5` method.